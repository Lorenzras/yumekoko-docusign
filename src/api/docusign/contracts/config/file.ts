import path from 'path';

/**
 * Config for contract generated folder
 */

export const assetsDir = path.join(__dirname, '../assets');
export const latestPDF = '請負契約書20221013.pdf';
export const latestXlsx = '請負契約書20221013.xlsx';
