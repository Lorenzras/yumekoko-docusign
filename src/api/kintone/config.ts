import {KintoneRestAPIClient} from '@kintone/rest-api-client';

const options = {
  baseUrl: process.env.KT_BASE_URL,
  auth: {
    apiToken: [
      process.env.KT_CUSTGROUPS,
      process.env.KT_PROJECTS,
      process.env.KT_PROJECT_ESTIMATES,
      process.env.KT_CUSTOMERS,
      process.env.KT_EMPLOYEES,
    ],
  },
};

console.log(process.env.node_env);

export enum APPIDS {
  customers = 173,
  stores = 19,
  employees = 34,
  custGroup = 185,
  custMemo = 181,
  constructionType = 190,
  projectDetails = 194,
  projEstimate = 202

}


export const KintoneClient = new KintoneRestAPIClient(options);

export const KintoneRecord = KintoneClient.record;

export default KintoneClient;
