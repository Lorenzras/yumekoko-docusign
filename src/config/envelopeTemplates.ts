export const templateIds = {
  test: '437e0841-600c-4fc3-8266-beddfbab1606',
  工事請負変更契約書: 'b46f9138-8e6b-49db-854b-e586172fb3a6',
  請負契約書: 'd12037c6-8c2c-473e-8237-d66c62705029',
  請負契約書Static: '1fc9846f-bb91-4385-a100-c5aa8ace7390',

};
