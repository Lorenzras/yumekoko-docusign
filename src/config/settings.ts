import fs from 'fs/promises';
import path from 'path';
import dotenv from 'dotenv';
import docusign from 'docusign-esign';


dotenv.config();


export const ds = docusign;
export const oAuth = ds.ApiClient.OAuth;
export const restApi = ds.ApiClient.RestApi;

export const basePath = restApi.BasePath.DEMO;
export const oAuthBasePath = oAuth.BasePath.DEMO;
export const redirectURI = 'http://localhost';
export const expiresIn = 60 * 60 * 8;
export const tokenReplaceMin = 10;


export const getPrivateKey = async () => {
  return await fs
    .readFile(
      path.join(__dirname, 'keys', 'private.key'),
    );
};

export const integratorKey = <string>process.env.DS_INTEGRATOR_KEY;
export const userId = <string>process.env.DS_USER_ID;
export const scopes = [
  oAuth.Scope.IMPERSONATION,
  oAuth.Scope.SIGNATURE,
];
export const apiClient = new ds.ApiClient({
  basePath: basePath,
  oAuthBasePath: oAuthBasePath,

});

export const getEnvelopesApi = () => new docusign.EnvelopesApi(apiClient);

console.log('Integrator key ' + integratorKey);

