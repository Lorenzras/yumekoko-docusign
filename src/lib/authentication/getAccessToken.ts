import {addSeconds, subMinutes} from 'date-fns';
import {
  getPrivateKey,
  integratorKey,
  userId,
  scopes,
  expiresIn,
  tokenReplaceMin,
  apiClient,
} from '../../config';

/**
 *
 * Async function to obtain a accessToken via JWT grant
 *
 * RETURNS {accessToken, tokenExpirationTimestamp}
 *
 * We need a new accessToken. We will use the DocuSign SDK's function.
 */

export const getAccessToken = async () => {
  // moment().add(results.body.expires_in, 's').subtract(tokenReplaceMin, 'm')

  const privateKey = await getPrivateKey();


  const results: IRequestJWTUserTokenResponse = await apiClient
    .requestJWTUserToken(
      integratorKey,
      userId,
      scopes,
      privateKey,
      expiresIn,
    );

  const expiresAt = subMinutes(
    addSeconds(new Date(), +results.body.expires_in),
    tokenReplaceMin,
  );

  apiClient.addDefaultHeader(
    'Authorization', `Bearer ${results.body.access_token}`,
  );


  return {
    accessToken: results.body.access_token,
    tokenExpirationTimestamp: expiresAt,
  };
};

/* DsJwtAuth.prototype.getToken = async function _getToken() {
    // Data used
    // dsConfig.dsClientId
    // dsConfig.impersonatedUserGuid
    // dsConfig.privateKey
    // dsConfig.dsOauthServer

    const jwtLifeSec = 10 * 60, // requested lifetime for the JWT is 10 min
        dsApi = new docusign.ApiClient();
    dsApi.setOAuthBasePath(dsConfig.dsOauthServer.replace('https://', '')); // it should be domain only.
    const results = await dsApi.requestJWTUserToken(dsConfig.dsClientId,
        dsConfig.impersonatedUserGuid, this.scopes, rsaKey,
        jwtLifeSec);

    const expiresAt = moment().add(results.body.expires_in, 's').
    subtract(tokenReplaceMin, 'm');
    this.accessToken = results.body.access_token;
    this._tokenExpiration = expiresAt;
    return {
        accessToken: results.body.access_token,
        tokenExpirationTimestamp: expiresAt
    };
} */
