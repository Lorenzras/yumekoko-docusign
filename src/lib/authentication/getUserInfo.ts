import {apiClient} from '../../config';
import {getAccessToken} from './getAccessToken';

/**
 * Get user info.
 *
 * @returns {IGetUserInfoResponse} returns Userinfo
 */
export const getUserInfo = async () => {
  const {accessToken} = await getAccessToken();
  const userInfo: IGetUserInfoResponse = await apiClient
    .getUserInfo(accessToken);


  return userInfo;
};
