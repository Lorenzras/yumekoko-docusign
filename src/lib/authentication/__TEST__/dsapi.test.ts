import {apiClient, ds, integratorKey, userId} from '../../../config';
import {getAccessToken} from '../getAccessToken';
import {getUserInfo} from '../getUserInfo';
describe('DS API', ()=>{
  it('should list folders', async () => {
    const {accounts} = await getUserInfo();
    const {accountId} = accounts[0];


    const api = new ds.FoldersApi(apiClient);


    const res = await api.list(accountId);

    expect(res).toMatchSnapshot();
  }, 1000*60);
});
