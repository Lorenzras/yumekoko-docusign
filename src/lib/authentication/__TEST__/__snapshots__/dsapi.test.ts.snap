// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DS API should list folders 1`] = `
exports {
  "endPosition": "3",
  "folders": Array [
    exports {
      "folderId": "09f4d726-70c3-425f-8552-6ee4b63f228d",
      "hasSubFolders": "false",
      "itemCount": "0",
      "name": "Draft",
      "owner": exports {
        "email": "lenzras@gmail.com",
        "userId": "9169dfa2-cb51-47e7-bf93-da59d152da90",
        "userName": "Lorenz Ras",
      },
      "subFolderCount": "0",
      "type": "draft",
      "uri": "/folders/09f4d726-70c3-425f-8552-6ee4b63f228d",
    },
    exports {
      "folderId": "b51337f0-59e2-4272-a125-95feb6fcf70b",
      "hasSubFolders": "false",
      "itemCount": "0",
      "name": "Inbox",
      "owner": exports {
        "email": "lenzras@gmail.com",
        "userId": "9169dfa2-cb51-47e7-bf93-da59d152da90",
        "userName": "Lorenz Ras",
      },
      "subFolderCount": "0",
      "type": "inbox",
      "uri": "/folders/b51337f0-59e2-4272-a125-95feb6fcf70b",
    },
    exports {
      "folderId": "5feaec83-0da4-4c91-9fad-4abbcf18152e",
      "hasSubFolders": "false",
      "itemCount": "0",
      "name": "Deleted Items",
      "owner": exports {
        "email": "lenzras@gmail.com",
        "userId": "9169dfa2-cb51-47e7-bf93-da59d152da90",
        "userName": "Lorenz Ras",
      },
      "subFolderCount": "0",
      "type": "recyclebin",
      "uri": "/folders/5feaec83-0da4-4c91-9fad-4abbcf18152e",
    },
    exports {
      "folderId": "25196255-511c-492a-8e9a-9c1a8dad5d5d",
      "hasSubFolders": "false",
      "itemCount": "2",
      "name": "Sent Items",
      "owner": exports {
        "email": "lenzras@gmail.com",
        "userId": "9169dfa2-cb51-47e7-bf93-da59d152da90",
        "userName": "Lorenz Ras",
      },
      "subFolderCount": "0",
      "type": "sentitems",
      "uri": "/folders/25196255-511c-492a-8e9a-9c1a8dad5d5d",
    },
  ],
  "resultSetSize": "4",
  "startPosition": "0",
  "totalSetSize": "4",
}
`;
